templateEngine/index.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+import nunjucks from 'nunjucks';
+import isFilePath from './utils/isFilePath/index.js';
+
+let environment;
+
+/**
+ * Factory function to create a Nunjucks template engine service.
+ *
+ * @param {object} [app] - Optional Express app instance to integrate with Nunjucks.
+ * @returns {object} Service object with methods to initialise and interact with the template engine.
+ */
```

Comment:
Test Coverage (not shown)

Recommend tests for:
- init idempotency.
- render on inline vs file path fallback.
- Global variables present in environment.
- Behavior when invalid template name passed (error surfaces).